### PR TITLE
[service] ensure otlp emitted logs respect severity

### LINKED ISSUE #11717: [service] logs emitted by collector don't follow severity
The collector supports emitting its internal logs via OTLP. After testing this functionality, it appears that logs of all severities are emitted via the logger provider configured. This is because the otelzap bridge doesn't have a configuration for severity level. This requires a separate processor ([minsev](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/processors/minsev)) which is currently not supported by the otel configuration schema.

I will submit a pull request that proposes wrapping the Tee'd zap core into another zap core that will apply the filtering before logs are sent to the logger provider.

### PR DESCRIPTION
**Pull request by codeboten**
Fixes #11717

### CODE REVIEW

File: cmd/builder/README.md [hunk 1]
```diff
@@ -126,7 +126,7 @@ Finally, load the OpenTelemetry Collector as a project in the IDE, configure deb
 The configuration file is composed of two main parts: `dist` and module types. All `dist` options can be specified via command line flags:
 
 ```console
-ocb --config=config.yaml --name="my-otelcol"
+ocb --config=config.yaml
```
Comment (TylerHelmuth) [inline cmd/builder/README.md:129]: Unrelated, but also needs updated

**Review by TylerHelmuth — APPROVED** [review]

File: .chloggen/codeboten_fix-11717.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: service
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: ensure otlp emitted logs respect severity
```
Comment (mx-psi) [inline .chloggen/codeboten_fix-11717.yaml:10]:
```suggestion
note: ensure OTLP emitted logs respect severity
```

File: package.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
   "devDependencies": {
-    "markdown-link-check": "3.12.2"
+    "markdown-link-check": "3.13.6"
```
Comment (codeboten) [inline package.json:3]:
```suggestion
    "markdown-link-check": "3.12.2"
```

File: cmd/builder/README.md (hunk 1 above)
Comment (codeboten) [reply cmd/builder/README.md]: fixed in a separate PR

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (codeboten) [reply package.json]: you're right, fixed

File: service/telemetry/logger.go [hunk 4]
```diff
@@ -42,12 +42,13 @@ func newLogger(set Settings, cfg Config) (*zap.Logger, log.LoggerProvider, error
 		lp = set.SDK.LoggerProvider()
 
 		logger = logger.WithOptions(zap.WrapCore(func(c zapcore.Core) zapcore.Core {
-			return zapcore.NewTee(
+			core, _ := zapcore.NewIncreaseLevelCore(zapcore.NewTee(
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply service/telemetry/logger.go:45]: It will panic later, but we won't see the error from zap, which could be useful for troubleshooting

**Review by mx-psi — APPROVED** [review]

**Pull request merged**